frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts

```diff
@@ -91,16 +95,45 @@ export class ImportExperimentComponent implements OnInit {
   }
   
   async uploadFile(event) {
+    let index = 0, fileName = '';
+    this.importFileErrors = [];
     const reader = new FileReader();
+
+    readFile(index);
+    function readFile(index: number) {
+      if (index >= event.target.files.length) return;
+      fileName = event.target.files[index].name;
+      reader.readAsText(event.target.files[index]);
+    }
+
     reader.addEventListener(
       'load',
       async function() {
         const result = JSON.parse(reader.result as any);
         this.experimentInfo = result;
         this.isExperimentJSONVersionValid = await this.validateExperimentJSONVersion(this.experimentInfo);
+        this.isExperimentJSONValid = this.validateExperimentJSON(this.experimentInfo);
+
+        if (this.isExperimentJSONVersionValid && this.isExperimentJSONValid) {
+          this.allExperiments.push(this.experimentInfo);
+        } else if (!this.isExperimentJSONValid) {
+          this.importFileErrors.push({
+            fileName: fileName,
+            error: this.translate.instant('home.import-experiment.error.message.text')
+          });
+        } else {
+          this.importFileErrors.push({
+            fileName: fileName,
+            error: this.translate.instant('home.import-experiment.version-error.message.text')
+          });
+          this.allExperiments.push(this.experimentInfo);
+        }
+
+        this.importFileErrorsDataSource.data = this.importFileErrors;
+        readFile(++index);
       }.bind(this)
     );
-    reader.readAsText(event.target.files[0]);
+    // reader.readAsText(event.target.files[0]);
```

Comment:
can this be removed instead of commented out?

Reply:
Removed this part